src/components/clusterWizard/ClusterDetails.tsx

```diff
@@ -271,7 +284,7 @@ const ClusterDetailsForm: React.FC<ClusterDetailsFormProps> = (props) => {
                 (dirty || (cluster && canNextClusterDetails({ cluster })))
               )
             }
-            onNext={submitForm}
+            onNext={handleOnNext(dirty, submitForm, cluster)}
```

Comment:
@mareklibra or @jtomasek in case you ask... :) Formik doesn't allow to read the dirty prop inside the `handleSubmit` callback we already have defined (upwise in this component)